### PR TITLE
Version 1 of Inventory (#19) for #3

### PR DESCRIPTION
**Pull request by mitsudz**
This is intended as the first implementation of an inventory (#19 ) for feature #3  - it implements an interface which can hopefully later be abstracted to represent all in-game containers and (fingers crossed) thoroughly tests the initial inventory implementation.

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/inventory/Inventory.java [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    }
+
+    /**
+     * Adds an item to the inventory, either by stacking it with existing items or placing it in
+     * a new slot.
+     *
+     * @param item the item to add to the inventory.
+     */
+    @Override
+    public void add(AbstractItem item) {
```
Comment (lellis0406) [inline source/core/src/main/com/csse3200/game/inventory/Inventory.java:186]: this should probably have a boolean return value to indicate whether pickup was successful; I expect items will behave differently if you try to pick them up and fail

**Review by lellis0406 — CHANGES_REQUESTED** [review]
looks good, see comment for single improvement, once that is addressed then should be good to go

**Comment by lhick2108** [comment]
I'm happy with everything, that I've looked through.

File: source/core/src/main/com/csse3200/game/inventory/Inventory.java (hunk 1 above)
Comment (mitsudz) [reply source/core/src/main/com/csse3200/game/inventory/Inventory.java]: added

**Pull request merged**